serial/serial_resources/process_entry.py

```diff
@@ -0,0 +1,149 @@
+import serial
+import os
+import mimetypes
+from datetime import datetime
+from ..api_client import APIClient 
+from .dataset import Datasets 
+
+class ProcessEntry:
+    def __init__(self, process_entry_data): 
+        """
+        Creates a new Process Entry
+        
+        Args:
+        - client: A Serial client object
+        - component_instance: Component Instance object 
+        - process_id: UUID for the process associated with this entry
+
+        Returns:
+        - A newly created process entry
+        """
+        self.client = APIClient(serial.api_key, serial.base_url) 
+        self.data = process_entry_data
+        self.process_id = process_entry_data["process_id"] 
+        self.id = process_entry_data["id"] 
+
+    def add_text(self, dataset_name, value, expected_value=None):
+        """
+        Add text data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - value: The value to be submitted
+        - expected_value: Optional argument to pass an expected value. If the 
+        values are the same, it would be considered a passing test result
+        """
+        if serial.debug:
+            print(f"Adding text data: {dataset_name} with value: {value} and expected value: {expected_value}")
+        dataset = serial.Datasets.get(dataset_name, "PARAMETRIC_QUALITATIVE", self.process_id)
+        data = {"type": "TEXT", "dataset_id": dataset.dataset_id, "value": value}
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+
+    def add_number(self, dataset_name, value, usl=None, lsl=None):
+        """
+        Add numerical data to a process entry
+
+        Args: 
+        - dataset_name: User facing name of the dataset
+        - value: The value to be submitted
+        - usl: Optional argument to override the dataset's upper spec limit
+        - lsl: Optional argument to override the dataset's lower spec limit
+        """
+        if serial.debug:
+            print(f"Adding numerical data: {dataset_name} with value: {value} and usl: {usl} & lsl: {lsl}")
+        dataset = serial.Datasets.get(dataset_name, "PARAMETRIC_QUANTITATIVE", self.process_id)
+        data = {"type": "NUMERICAL", "dataset_id": dataset.dataset_id, "value": value}
+        if usl:
+            data["usl"] = usl
+        if lsl:
+            data["lsl"] = lsl
+        return self.client.make_api_request(f"/processes/entries/{self.id}", "PUT", data=data)
+
+    def add_file(self, dataset_name, path, bucket_type, file_name=None):
```

Comment:
This is actually ready for the merging of images & files, we just have to change the meaning of bucket_type :)